src/move/market_metadata/sources/emojicoin_dot_fun_market_metadata.move

```diff
@@ -0,0 +1,624 @@
+module market_metadata::emojicoin_dot_fun_market_metadata {
+
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use aptos_std::smart_table::{Self, SmartTable};
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::{String};
+    use emojicoin_dot_fun::emojicoin_dot_fun;
+
+    /// Signer does not correspond to admin.
+    const E_NOT_ADMIN: u64 = 0;
+    /// Admin to remove address does not correspond to admin.
+    const E_ADMIN_TO_REMOVE_IS_NOT_ADMIN: u64 = 1;
+    /// Admin is already an admin.
+    const E_ALREADY_ADMIN: u64 = 2;
+    /// Admin to remove is market_metadata publisher.
+    const E_ADMIN_TO_REMOVE_IS_MARKET_METADATA_PUBLISHER: u64 = 3;
+    /// Market not found.
+    const E_MARKET_NOT_FOUND: u64 = 4;
+    /// The number of provided keys and values do not match.
+    const E_KEYS_AND_VALUES_NOT_EQUAL_LENGTH: u64 = 5;
+
+    struct Nil {}
+    has copy, drop, store;
+
+    struct Registry has key {
+        /// Addresses of signers who can mutate the vault.
+        admins: vector<address>,
+        /// Table of markets to market properties.
+        map: SmartTable<address, SimpleMap<String, String>>
+    }
+
+    #[view]
+    public fun admins(): vector<address> acquires Registry {
+        Registry[@market_metadata].admins
+    }
+
+    #[view]
+    public fun market_properties(
+        market: address
+    ): Option<SimpleMap<String, String>> acquires Registry {
+        let map =
+            &Registry[@market_metadata].map;
```

Comment:
```suggestion
        let map = &Registry[@market_metadata].map;
```

Reply:
Addressed in f3362f52f8c192548676036fcdfcc3312dc15162.